### DISCUSSION TITLE
Unable to find composer despite being available on path in Windows

### DISCUSSION THREAD

**Discussion by negentropicdev** [symfony/symfony#49338] [Q&A]
I'm not sure how symfony cli is checking for the existence of composer, I did a manual install of composer and have a composer.cmd that runs the full php command and I can simply type `composer --version` on the command-line and get a proper response.

This means the symfony cli continually redownloads its own copy of composer to a hidden folder in my user directory.

Is the symfony cli being too smart and doing more than just trying to run `composer --version` to check its existence?

**Comment by excitedbox** [comment]
I have had my own problems with updating symfony because it has changed the symfony cli and switched from composer to scoop so there have been multiple hidden config files left over in different locations that were either partially or fully overriding settings. I had Symfony Cli version conflicts where it would either fail to update or not use the new version when using certain commands, php version changing at random. Also caching issues can cause problems.

All these package management systems and clis such as npm, yarn, composer and symfony seem to create a mess over time, due to each using different config systems and making changes that leave trash from old versions to override configs. Npm is the worst in my opinion and I have considered doing a complete fresh install of all these systems just to get a clean start but there is always some hidden config file that now suddenly steals control.

I have lost weeks of my life to cleaning up this mess. 2 times this year I have spent 3 days just fixing conflicts and repository config conflicts.

**Comment by negentropicdev** [comment]
They spent the time to implement fallback behavior of downloading a fresh copy of composer, would be nice if it just failed and gave instructions on where things got broken.
